### PR TITLE
Group GSR Invite Cell

### PR DESCRIPTION
**Pull request by sanieldalib**
Added cell to the homepage for group gsr invites. A request is made to student-life server only if the gsrGroupsEnabled header is set to true to prevent unnecessary requests on prod for time being.

### CODE REVIEW

File: server/groups.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    if account:
+        x_authorization = request.headers.get('X-Authorization')
+        authorization = request.headers.get('Authorization')
+
+        headers = {'Authorization': authorization if authorization else x_authorization}
+        invite_url = 'https://studentlife.pennlabs.org/users/{}/invites'.format(account.pennkey)
+        r = requests.get(url=invite_url, headers=headers)
+        json = r.json()
+        return json
+    return list()
```
Comment (joshdoman) [inline server/groups.py]: Let's move this file to `server/studyspaces/groups.py`

File: server/homepage.py [hunk 2]
```diff
@@ -253,3 +260,11 @@ def get_data():
 
     td = datetime.timedelta(days=1)
     return cache_get('ios_version', td, get_data)
+
+
+def get_group_invite_cell(account):
+    invites = get_invites_for_account(account)
+    if invites:
+        return HomeCell('invites', invites, 1001)
+
```
Comment (joshdoman) [inline server/homepage.py]: Let's remove this blank line.

File: server/groups.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import requests
+from flask import request
+
+
+def get_invites_for_account(account):
+    if account:
```
Comment (joshdoman) [inline server/groups.py]: No need to check if account is null. Account is provided as a parameter and must therefore be non-null.

File: server/groups.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        x_authorization = request.headers.get('X-Authorization')
+        authorization = request.headers.get('Authorization')
+
+        headers = {'Authorization': authorization if authorization else x_authorization}
+        invite_url = 'https://studentlife.pennlabs.org/users/{}/invites'.format(account.pennkey)
+        r = requests.get(url=invite_url, headers=headers)
+        json = r.json()
+        return json
```
Comment (joshdoman) [inline server/groups.py]: If both the authorization header and x-authorization is null, we should not try even try to make a request. Instead, we should return null.

File: server/groups.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import requests
+from flask import request
+
+
+def get_invites_for_account(account):
```
Comment (joshdoman) [inline server/groups.py]: Let's add documentation for exactly what this function does and what it returns.

File: server/groups.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    if account:
+        x_authorization = request.headers.get('X-Authorization')
+        authorization = request.headers.get('Authorization')
+
+        headers = {'Authorization': authorization if authorization else x_authorization}
+        invite_url = 'https://studentlife.pennlabs.org/users/{}/invites'.format(account.pennkey)
```
Comment (joshdoman) [inline server/groups.py]: Why do we need to provide the user's pennkey? The access token should be enough to determine who they are. Also, isn't it security issue if a user can provide their access token but use someone else's pennkey to get their invites/groups? (This is question about the gsr server, shouldn't hold up this request).

**Comment by sanieldalib** [comment]
made needed changes!

**Pull request merged**